SAK-49631 - "Exit Student View" not working on other tabs except "Overview" tab

jonespm opened the pull request:
Some ideas for review.

I'm going to leave this as draft for now and am happy if someone does this differently or if this is almost good and I can fix it up.

Some things I didn't like:

I don't think these dropdowns should appear on the overview (iframe) divs but I don't see a better way to detect this other than `$pageColumnLayout == 'col1'`

I'm also not sure why some of these negative margins were in there and they made the link no longer appear. 

I'm not sure if there's any other edge cases here or if there's a better place this dropdown could be placed.

jonespm commented:
I probably should look at that `viewAsStudentLink` variable in the backend to avoid the frontend check.

kunaljaykam commented:
> I don't think these dropdowns should appear on the overview (iframe) divs but I don't see a better way to detect this other than `$pageColumnLayout == 'col1'`
> 
> I'm also not sure why some of these negative margins were in there and they made the link no longer appear.
> 
> I'm not sure if there's any other edge cases here or if there's a better place this dropdown could be placed.

I said the same thing to myself when I did this PR for the updated portal. I hope someone will tell me if there's a better way to do it :). I like it. This is better than what I did.  Thanks @jonespm

Do you mean, The Overview tool shouldn't have the option for role-swap? 

I think the negative margins were there to align the dropdown. I wasn't happy with it either.

kunaljaykam commented on library/src/skins/default/src/sass/modules/tool/portal/_portal.scss:
```diff
@@ -130,22 +130,15 @@
 }
 
+.role-switcher {
+    margin-right: 5px;
```
To align with the `link-button`
```suggestion
    margin: 2px 5px;
```

jonespm commented:
Thanks for the comments, I added your change and also changed it to check for isHomePage rather than the double column which seems more straightforward since this is the page with 99% of the issue. I think it's looking pretty nice now. I checked it also on mobile view and seems pretty nice there too.

![image](https://github.com/sakaiproject/sakai/assets/27447/7cc223b1-450b-4c64-9ca5-03e6c57a8aba)

kunaljaykam commented:
Thanks! with your fix. I think we can clear some CSS and JS as well. I have created a pull request on your branch for the same. Please take a look: https://github.com/jonespm/sakai/pull/2

kunaljaykam reviewed (approved):
LGTM! Thanks @jonespm
